### PR TITLE
Completion of JavassistParameterDeclaration

### PR DESCRIPTION
**Pull request by kgeilmann**
This PR completes the implementation of JavassistParameterDeclaration. 
For the isXYZ methods it copies the behaviour of ReflectionParameterDeclaration and if available the parameter name gets read from the jar.

### CODE REVIEW

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java [hunk 1]
```diff
@@ -206,4 +203,19 @@ private static ResolvedType typeArgumentToType(SignatureAttribute.TypeArgument t
...
+        CodeAttribute codeAttribute = methodInfo.getCodeAttribute();
+        if (codeAttribute != null) {
+            LocalVariableAttribute attr = (LocalVariableAttribute) codeAttribute.getAttribute(LocalVariableAttribute
+                    .tag);
+            if (attr != null) {
+                int pos = Modifier.isStatic(method.getModifiers()) ? 0 : 1;
+                return attr.variableName(paramNumber + pos);
+            }
+        }
+        return null;
```
Comment (ftomassetti) [inline javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java]: We should explain while this return null

File: javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistParameterDeclarationTest.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    private TypeSolver typeSolver;
+
+    @Before
+    public void setup() throws IOException {
+        Path pathToJar = adaptPath("src/test/resources/javaparser-core-3.0.0-alpha.2.jar");
+        typeSolver = new CombinedTypeSolver(new JarTypeSolver(pathToJar), new ReflectionTypeSolver());
+    }
+
+    @Test
+    public void noNamesAvailable_inInterfaceMethods() {
```
Comment (ftomassetti) [inline javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistParameterDeclarationTest.java]: I think we should not use underscores in names

**Review by ftomassetti — APPROVED** [review]
To me it seems good and a very valuable contribution: thanks. I have just minor comments

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java (hunk 1 above)
Comment (matozoid) [reply javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java]: Nothing in JavaParser should return null really. Use Optional.

**Comment by kgeilmann** [comment]
Good points. I've added a commit addressing your comments.

**Comment by matozoid** [comment]
Thanks!

**Pull request merged**